Plugins/Flow.Launcher.Plugin.Program/Main.cs

```diff
@@ -72,22 +72,17 @@ public List<Result> Query(Query query)
             Win32[] win32;
             UWP.Application[] uwps;
 
-            lock (IndexLock)
-            { // just take the reference inside the lock to eliminate query time issues.
-                win32 = _win32s;
-                uwps = _uwps;
-            }
-
-
-            var results1 = win32.AsParallel()
-                .Where(p => p.Enabled)
-                .Select(p => p.Result(query.Search, _context.API));
+            win32 = _win32s;
+            uwps = _uwps;
 
-            var results2 = uwps.AsParallel()
-                .Where(p => p.Enabled)
-                .Select(p => p.Result(query.Search, _context.API));
+            var result = win32.Cast<IProgram>()
+                 .Concat(uwps)
+                 .AsParallel()
+                 .Where(p => p.Enabled)
+                 .Select(p => p.Result(query.Search, _context.API))
+                 .Where(r => r != null && r.Score > 0)
+                 .OrderBy(r=>r.Score).ToList();
 
-            var result = results1.Concat(results2).Where(r => r != null && r.Score > 0).ToList();
```

Comment:
Since we have the IProgram interface, why not use it to make the query done in one parallel linq selection.